cluster-autoscaler/cloudprovider/gce/templates.go

```diff
@@ -332,17 +332,37 @@ func parseKubeReserved(kubeReserved string) (apiv1.ResourceList, error) {
 }
 
 func extractLabelsFromKubeEnv(kubeEnv string) (map[string]string, error) {
-	labels, err := extractFromKubeEnv(kubeEnv, "NODE_LABELS")
+	// In v1.10+, labels are only exposed for the autoscaler via AUTOSCALER_ENV_VARS
+	// see kubernetes/kubernetes#61119. We try AUTOSCALER_ENV_VARS first, then
+	// fall back to the old way.
+	labels, err := extractAutoscalerVarFromKubeEnv(kubeEnv, "node_labels")
 	if err != nil {
-		return nil, err
+		glog.Infof("node_labels not found via AUTOSCALER_ENV_VARS, will try NODE_LABELS")
+		labels, err := extractFromKubeEnv(kubeEnv, "NODE_LABELS")
+		if err != nil {
+			return nil, err
+		}
+		return parseKeyValueListToMap(labels)
```

Comment:
nit: you could remove this line and just exit the if block and let the line below return the map to reduce code duplication. up to you whether you think that's more confusing.